### PR TITLE
chore: add strict to Compose, Lima, Kubectl, KubeContext and Registries extensions

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

This is one of the PRs to complete https://github.com/containers/podman-desktop/issues/6374
To enable the `prefer-nullish-coalescing` rule we have to enable the `strictNullChecks`. 
This PR just covers  Compose, Lima, Kubectl and KubeContext extensions. 

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

part of #6374 

### How to test this PR?

N/A

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by benoitf — APPROVED** [review]

**Review by cdrage — APPROVED** [review]

File: extensions/compose/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "strictNullChecks": true,
```
Comment (jeffmaury) [inline extensions/compose/tsconfig.json]: strict would be prefered as it enable other checks as well

Comment (lstocchi) [reply extensions/compose/tsconfig.json]: updated

**Review by axel7083 — APPROVED** [review]
Nice ! I dream one of one having strict: true everywhere ahah !

File: extensions/kubectl-cli/types/shell-path.d.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**********************************************************************
```
Comment (axel7083) [inline extensions/kubectl-cli/types/shell-path.d.ts:1]: maybe it should in the `types` folder at the root ?

File: extensions/registries/tsconfig.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "strictNullChecks": true,
```
Comment (axel7083) [inline extensions/registries/tsconfig.json]: why this one it not `strict` ? too much changes ?

File: extensions/kubectl-cli/types/shell-path.d.ts (hunk 2 above)
Comment (benoitf) [reply extensions/kubectl-cli/types/shell-path.d.ts:1]: extensions need to be 'independent'

File: extensions/compose/src/compose-wrapper-generator.ts [hunk 4]
```diff
@@ -44,7 +44,7 @@ export class ComposeWrapperGenerator {
     }
 
     // render the template
-    return mustache.render(template, { socketPath, binFolder: this.binFolder });
+    return render(template, { socketPath, binFolder: this.binFolder });
```
Comment (benoitf) [inline extensions/compose/src/compose-wrapper-generator.ts]: looks like it's failing using that syntax

File: extensions/registries/tsconfig.json (hunk 3 above)
Comment (lstocchi) [reply extensions/registries/tsconfig.json]: Ops, i forgot it. Updated, thanks!! 👍

**Review by jeffmaury — APPROVED** [review]
Mustache requires an update, otherwise LGTM

File: extensions/compose/src/compose-wrapper-generator.ts (hunk 4 above)
Comment (lstocchi) [reply extensions/compose/src/compose-wrapper-generator.ts]: maybe we should downgrade mustache https://github.com/janl/mustache.js/issues/797 👀

Comment (lstocchi) [reply extensions/compose/src/compose-wrapper-generator.ts]: I haven't found a nice way to make it work. As it works at runtime and it seems there is an error in the typings i just added a command to ignore the error. If you have any suggestion to make it another way let me know

Comment (jeffmaury) [reply extensions/compose/src/compose-wrapper-generator.ts]: Is the change mandatory ?

Comment (lstocchi) [reply extensions/compose/src/compose-wrapper-generator.ts]: i guess so, it complains that there are no mustache types if i leave the code as it was... but then if i add the types they are missing the default export

**Review by benoitf — APPROVED** [review]
would be interesting to have a better solution for mustache but can be a follow up

**Pull request merged**